src/scanner.mll

```diff
@@ -0,0 +1,103 @@
+(* Ocamllex scanner for Sift *)
+
+{ open Parser }
+
+let digit = ['0' - '9']
+let digits = digit+
+let exponent = ['e' 'E'] ['+' '-']? digits
+
+rule token = parse
+  [' ' '\t' '\r' '\n'] { token lexbuf } (* Whitespace *)
+| "/*"     { comment lexbuf }           (* Comments *)
+(* parentheses and brackets *)
+| '(' { LPAREN }
+| ')' { RPAREN }
+| '{' { LBRACE }
+| '}' { RBRACE }
+| '[' { LSQBRACE }
+| ']' { RSQBRACE }
+| '<' { LARROW }
+| '>' { RARROW }
+
+(* arithmetic operators *)
+| '+' { ADD }
+| '-' { SUBTRACT }
+| '*' { MULTIPLY }
+| '/' { DIVIDE }
+| '^' { POWER }
+| "%" { MOD }
+
+(* relational operators *)
+| "==" { EQ }
+| ">=" { GEQ }
+| '>' { GT }
+| "<=" { LEQ }
+| '<' { LT }
+| "!=" { NEQ }
+
+(* delimiters *)
+| ',' { COMMA }
+| ':' { COLON }
+| ';' { SEMICOLON }
+| '.' { DOT }
+
+(* assignment *)
+| "=" { ASSIGN }
+
+(* pipe *)
+| "|>" { PIPE }
+
+(* logical operators *)
+| "&&" { AND }
+| "||" { OR }
+| '!' { NOT }
+| '^' { XOR }
+(* Keywords *)
+| "if" { IF }
+| "elif" { ELIF }
```

Comment:
Are we supporting `elif`?

Reply:
Same comment as in parser. Easy to tokenize than else if